### PR TITLE
Feature/cipher

### PR DESCRIPTION
**Pull request by aloiscochard**
Add a few simple functions to work with `Cipher` on `BitVector` and `ByteVector`.

### CODE REVIEW

File: core/src/main/scala/scodec/bits/BitVector.scala [hunk 1]
```diff
@@ -1013,6 +1016,10 @@ sealed trait BitVector extends BitwiseOperations[BitVector, Long] with Serializa
    */
   final def digest(digest: MessageDigest): BitVector = BitVector(bytes.digest(digest))
 
+  final def cipher(ci: Cipher, key: SecretKeySpec, opmode: Int): BitVector = BitVector(bytes.cipher(ci, key, opmode))
```
Comment (mpilquist) [inline core/src/main/scala/scodec/bits/BitVector.scala]: Can we make this method private? Perhaps this is a silly concern, but I don't like the way JCA uses an Int to represent the opmode.

Comment (aloiscochard) [inline core/src/main/scala/scodec/bits/BitVector.scala]: Agree!

File: core/src/main/scala/scodec/bits/BitVector.scala [hunk 2]
```diff
@@ -1013,6 +1016,10 @@ sealed trait BitVector extends BitwiseOperations[BitVector, Long] with Serializa
...
+  final def encrypt(ci: Cipher, key: SecretKeySpec): BitVector = cipher(ci, key, Cipher.ENCRYPT_MODE)
```
Comment (mpilquist) [inline core/src/main/scala/scodec/bits/BitVector.scala]: Can you change `SecretKeySpec` to `Key`? That will let this work with more algorithms, including asymmetric ones.

Comment (mpilquist) [inline core/src/main/scala/scodec/bits/BitVector.scala]: On other issue -- I think we should support initializing the cipher using all 4 params (opmode, key, algorithm params, and secure random). Not sure what this should look like in the API though. Default arguments is one option. I think this concern is what led to the creation of `CipherFactory` in scodec-core.

**Comment by aloiscochard** [comment]
@mpilquist all comments applied, thanks!

**Comment by aloiscochard** [comment]
@mpilquist I have now added `AlgorithmParameters` param that I forgot before, should be fine now.

**Comment by mpilquist** [comment]
:+1: Thanks for this!

Do you want to add ScalaDoc with `@group conversions`? Doc is rather unnecessary though I like to have every method documented in scodec-bits. I can do it after merging though too -- up to you.

Oh it might be good in the docs to mention that any exceptions thrown by the cipher are propagated -- I think that's a reasonable expectation for folks used to JCA but for folks that haven't used JCA, they might be surprised to see exceptions.

**Pull request merged**